nixos/doc/manual/release-notes/rl-2405.section.md

```diff
@@ -516,6 +516,8 @@ The pre-existing [services.ankisyncd](#opt-services.ankisyncd.enable) has been m
 - `erlang_node_short_name`, `erlang_node_name`, `port` and `options` configuration parameters are gone, and have been replaced with an `environment` parameter.
     Use the appropriate [environment variables](https://hexdocs.pm/livebook/readme.html#environment-variables) inside `environment` to configure the service instead.
 
+- `postgresql_12` will be EOL on Nov. 14 2024 which is within the lifetime of 24.05. Thus it has been dropped.
+
```

Comment:
Would PostgreSQL 12 receive security fixes between 2024-11-15 and 2025-01-01? If not, then we should drop it.

Reply:
> I'd argue for everyone who still wants to use v12 its much more of a problem security-wise if they stop updating nixpkgs for half a year instead

I disagree with that conclusion: with that reasoning we could keep virtually any unmaintained software in nixpkgs to allow users to upgrade.

> If v12 is dropped we are effectively blocked from updating nixpkgs. It's not easily possible to add an override for v12, because I don't know how to deal with that removed patch, which was specific for v12, in an override.

Fwiw there are at least two approaches here:
* using nixos-23.11 for just postgresql_12
* vendor the patch and the expression

---

But given that policy you do have a point.

An alternative would be to say we keep postgresql_12, but we set clear expectations: if anything bad happens after Nov 14 (i.e. the last minor it'll get) and I can't find the patch very quickly / can't apply it w/o conflicts it'll be marked as insecure immediately.

Not entirely sure yet, but I think that seems good. We should document it though.

Thoughts @mweinelt @NixOS/nixos-release-managers ?

Reply:
Fine with me if the follow through on this.